core/download.ts

```diff
@@ -40,6 +52,30 @@ function detectType(text: string): string | undefined {
   }
 }
 
+function detectEncodeType(response: Partial<Response>): string {
```

Comment:
I am hot sure that allow-list is the best option here since we will be limited to this list of encodings.

Maybe it is better to parse enconding from `headers.get('content-type')` string?

Like `/;\s*charset=(\w+)`

Reply:
thx. updated